packages/adapter-components/src/definitions/user/fetch_config.ts

```diff
@@ -95,7 +105,8 @@ export const createUserFetchConfigType = ({
     fetchEntryType.fields.criteria = new Field(fetchEntryType, 'criteria', fetchCriteriaType)
   }
 
-  const elemIDPartType = createMatchingObjectType<Omit<FieldIDPart, 'condition' | 'custom'>>({
+  // TODO: find if there's a way to specify the generic type here of TCustomNameMappingOptions
+  const elemIDPartType = createMatchingObjectType<Omit<FieldIDPart<NameMappingOptions>, 'condition' | 'custom'>>({
```

Comment:
Left this comment so we can discuss it during the review -
For some reason, when I specify the generic type as:

```suggestion
  // TODO: find if there's a way to specify the generic type here of TCustomNameMappingOptions
  const elemIDPartType = createMatchingObjectType<Omit<FieldIDPart<TCUstomNameMappingOptions>, 'condition' | 'custom'>>({
```

The resulting type causes an unexpected issue with `fieldName` under `fields`: TypeScript expects its `_annotations: { required: ?}` field to simultaneously satisfy contradictory conditions (both falsy and truthy) 😐. I've attempted to resolve this without success.
It seems like TypeScript is not accurately recognizing which elements should be considered optional, leading to confusion in determining the appropriate `SaltoAnnotationsForField` type for this field.

Reply:
weird, gonna need to look more later - either way since we currently don't enforce the enum values it's not critical :)